lib/backend/k8s/resources/customresource.go

```diff
@@ -47,6 +47,13 @@ type customK8sResourceClient struct {
 	k8sListType         reflect.Type
 	namespaced          bool
 	resourceKind        string
+	versionconverter    VersionConverter
+}
+
+// VersionConverter converts v1 or v3 k8s resources into v3 resources.
+// For a v3 resource, it's a no-op
```

Comment:
Wording is a little off.  More like "For a v3 resource, the conversion should be a no-op"